### PR TITLE
tools: build `kustomize` from the `vendor/` directory

### PR DESCRIPTION
**Pull request by nixpanic**
By installing `kustomize` from the vendor directory, versions of the
tools can automatically be updated, just like other dependencies. It
also makes things easier to install the tools in environments that have
slow networking or are completely disconnected.

### CODE REVIEW

File: tools.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+//go:build tools
+// +build tools
+
+/*
+Copyright 2021 The Kubernetes-CSI-Addons Authors.
```
Comment (Madhu-1) [inline tools.go:5]: 2022

File: Makefile [hunk 2]
```diff
@@ -43,11 +43,15 @@ help: ## Display this help.
...
-	$(CONTROLLER_GEN) rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+	$(CONTROLLER_GEN) rbac:roleName=manager-role crd webhook paths="./cmd/... ./controllers/... ./sidecar/..." output:crd:artifacts:config=config/crd/bases
 
 .PHONY: generate
 generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
-	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./api/..."
+
+.PHONY: bundle
+bundle: kustomize operator-sdk
```
Comment (Madhu-1) [inline Makefile:53]: can we have CI to validate kustomize/manifests/generate commands to make sure it's working and won't break in the future?

File: tools.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+// These imports are used to maintain a copy of the used tools in the vendor/
+// directory. Versions of the tools can automatically be updated, just like
+// other dependencies. It also makes things easier to install the tools in
+// environments that have slow networking or are completely disconnected.
+
+package main
+
+import (
+	_ "sigs.k8s.io/kustomize/kustomize/v4"
```
Comment (Madhu-1) [inline tools.go:28]: if kustomize is updated to v5 we need changes here also. will dependent bot takes care of it?

**Comment by deepshikhaaa** [comment]
+1, It will be helpful for downstream builds too.

File: Makefile (hunk 2 above)
Comment (nixpanic) [reply Makefile:53]: I hope so, with this change it should be easier to write such jobs.

Actually, the `bundle` target is not intentionally included, I am only experimenting with that now.

File: tools.go (hunk 1 above)
Comment (nixpanic) [reply tools.go:5]: right!

File: tools.go (hunk 3 above)
Comment (nixpanic) [reply tools.go:28]: no, I do not think so. But The way it is currently done (version in the `Makefile`) will not even see minor updates.

**Review by Madhu-1 — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Pull request merged**